src/App.tsx

```diff
@@ -1,95 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [total, setTotal] = useState(9);
+
+  const lastElemetOnPage = (currentPage * itemsPerPage) <= items.length
+    ? currentPage * itemsPerPage
+    : (currentPage - 1) * itemsPerPage + (items.length % itemsPerPage);
+
+  const firstElemetOnPage = (currentPage - 1) * itemsPerPage;
+
+  useEffect(() => {
+    setTotal(Math.ceil(items.length / itemsPerPage));
+  });
+
+  const handleChangeCurrentPage = (
+    event: React.FormEvent<HTMLAnchorElement>,
+  ) => {
+    const page = event.currentTarget.href.split('#').slice(-1)[0];
+
+    if (page === 'prev') {
+      setCurrentPage(currentPage - 1);
+    } else if (page === 'next') {
+      setCurrentPage(currentPage + 1);
+    } else if (
+      currentPage !== Number(page)
+      && currentPage >= 1
+      && currentPage <= total
+    ) {
+      setCurrentPage(Number(page));
+    }
```

Comment:
Optional advice, try not to use if/else statement
https://levelup.gitconnected.com/how-you-can-avoid-using-else-in-your-code-871197a1adbc